models/classes/task/helper/TaskIterator.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2020 (original work) Open Assessment Technologies SA;
+ */
+
+declare(strict_types=1);
+
+namespace oat\tao\model\task\helper;
+
+use common_persistence_sql_QueryIterator;
+use Doctrine\DBAL\Connection;
+use Iterator;
+use oat\generis\model\OntologyAwareTrait;
+use oat\generis\model\OntologyRdf;
+use oat\oatbox\service\ConfigurableService;
+
+class TaskIterator extends ConfigurableService
```

Comment:
```suggestion
class StatementTaskIterator extends ConfigurableService
```

I know I put this name `TaskIterator` as example, but I agreed with you to use the statement as part of class name in this case.